src/lualib/declarations/global.d.ts

```diff
@@ -10,6 +10,7 @@ declare function type(
     value: any
 ): "nil" | "number" | "string" | "boolean" | "table" | "function" | "thread" | "userdata";
 declare function setmetatable<T extends object>(table: T, metatable: any): T;
+declare function getmetatable<T extends object>(table: T): any;
```

Comment:
I'm considering adding a second type argument, ie ` TInput` and `TMetaTable` and having this return typed as ` `TMetaTable`, thoughts?

Reply:
What do you think about using lua-types in lualib? I have originally wanted to avoid it as it may slow down some PRs and we haven't used standard library too much, but now I think it may give more advantages making lualib more sound

Reply:
About to suggest the same thing. I think that makes much more sense to have common, well-maintained definitions than to partially reimplement them for Lualib.

Reply:
I think that's probably out of scope for this PR, but we should discuss it. I'm currently leaning to it's alright, but maybe we should consider moving lua-types to the TypeScriptToLua org in that case.